File: program/src/state.rs
use borsh::{BorshSerialize, BorshDeserialize};
use solana_program::pubkey::Pubkey;

#[derive(BorshSerialize, BorshDeserialize, Debug, Clone)]
pub struct Blog {
    pub authority: Pubkey,
    pub bump: u8,
    pub post_count: u8 // 10 posts max
}

#[derive(BorshSerialize, BorshDeserialize, Debug, Clone)]
pub struct Post {
    pub author: Pubkey,
    pub blog: Pubkey,
    pub bump: u8,
    pub slug: String,
    pub title: String,
    pub content: String,
}

impl Blog {
    pub const LEN: usize = 32 + 1 + 1;
}

impl Post {
    pub const LEN: usize = 32 + 32 + 1 + 10 + 20 + 50;
}

File: scripts/src/state.ts
import { PublicKey } from "@solana/web3.js";
import * as borsh from "@project-serum/borsh";

export interface BlogLayout {
  authorityPubkey: PublicKey;
  bump: number;
  postCount: number;
}

export interface PostLayout {
  slug: string;
  title: string;
  content: string;
}

export const BLOG_ACCOUNT_DATA_LAYOUT = borsh.struct([
  borsh.publicKey("authorityPubkey"),
  borsh.u8("bump"),
  borsh.u8("postCount"),
]);

export const POST_ACCOUNT_DATA_LAYOUT = borsh.struct([
  borsh.publicKey("author"),
  borsh.publicKey("blog"),
  borsh.u8("bump"),
  borsh.str("slug"),
  borsh.str("title"),
  borsh.str("content"),
]);

export const IX_DATA_LAYOUT = borsh.struct([
  borsh.u8("variant"),
  borsh.str("slug"),
  borsh.str("title"),
  borsh.str("content"),
]);
